criu/kerndat.c

```diff
@@ -1060,19 +1062,51 @@ static int kerndat_has_openat2(void)
 	return 0;
 }
 
-#define KERNDAT_CACHE_FILE     KDAT_RUNDIR "/criu.kdat"
-#define KERNDAT_CACHE_FILE_TMP KDAT_RUNDIR "/.criu.kdat"
+#define KERNDAT_CACHE_NAME	"criu.kdat"
+#define KERNDAT_CACHE_FILE	KDAT_RUNDIR"/"KERNDAT_CACHE_NAME
+
+static int get_kerndat_filename(char **kdat_file)
+{
+	int ret;
+
+	/*
+	 * Running as non-root, even with CAP_CHECKPOINT_RESTORE, does not
+	 * allow to write to KDAT_RUNDIR which usually is only writable by root.
+	 * Let's write .criu.kdat file to the home directory for non-root cases.
+	 */
+	if (opts.uid) {
```

Comment:
nit: opts.unprivileged

Reply:
Fixed in ef9d3a0b6a5a31e3f63a7a1d4d68fb8796c1e26d as suggested, thanks.